plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,824 @@
+        payload["ImportOptions"] = "Force"
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __import_license_http(self, module, import_license_url, resource_id):
+        payload = {}
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        proxy_details = get_proxy_details(module)
+        payload.update(proxy_details)
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __import_license_cifs(self, module, import_license_url, resource_id):
+        payload = {}
+        payload["ShareType"] = "CIFS"
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        if module.params.get('share_parameters').get('workgroup'):
+            payload["Workgroup"] = module.params.get('share_parameters').get('workgroup')
+        share_details = get_share_details(module)
+        payload.update(share_details)
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __import_license_nfs(self, module, import_license_url, resource_id):
+        payload = {}
+        payload["ShareType"] = "NFS"
+        payload["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+        payload["ShareName"] = module.params.get('share_parameters').get('share_name')
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __check_file_extension(self, module):
+        share_type = module.params.get('share_parameters').get('share_type')
+        file_name = module.params.get('share_parameters').get('file_name')
+        valid_extensions = {".txt", ".xml"} if share_type == "local" else {".xml"}
+        file_extension = file_name.lower().endswith(tuple(valid_extensions))
+        if not file_extension:
+            module.exit_json(msg=INVALID_FILE_MSG, failed=True)
+
+    def __get_import_license_url(self, module):
+        uri, error_msg = validate_and_get_first_resource_id_uri(
+            self.module, self.idrac, MANAGERS_URI)
+        if error_msg:
+            self.module.exit_json(msg=error_msg, failed=True)
+        resp = get_dynamic_uri(self.idrac, uri)
+        url = resp.get('Links', {}).get(OEM, {}).get(MANUFACTURER, {}).get(LICENSE_MANAGEMENT_SERVICE, {}).get('@odata.id', {})
+        action_resp = get_dynamic_uri(self.idrac, url)
+        if module.params.get('share_parameters').get('share_type') == "local":
```

Comment:
You can use ternary operator to reduce if else statement, try that. Check other places too.

Reply:
Done at all applicable places